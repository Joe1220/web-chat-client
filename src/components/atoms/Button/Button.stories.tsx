import React from 'react'
import { storiesOf } from '@storybook/react'
import { text, boolean } from '@storybook/addon-knobs'

import Button from './'

const stories = storiesOf('Components/Atoms', module)

stories
  .add('Button', () => (
    <Button lg={boolean('large button', false)}
            full={boolean('full width button', false)}
            grey={boolean('grey button', false)}
            onClick={() => null}
    >
      {text('text', 'test')}
    </Button>
  ) 
)

export default undefined