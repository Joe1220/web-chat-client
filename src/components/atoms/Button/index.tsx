import styled, { css } from 'styled-components'

interface IButton {
  lg?: boolean
  grey?: boolean
  full?: boolean
  onClick?: (e: any) => void
} 

const Button = styled('button')<IButton>`
  display: inline-block;
  font-size: ${props => props.theme.fontSize.base};
  justify-content: center;
  color: ${props => props.theme.colors.white};
  padding: ${props => props.theme.spaces.buttonPadding};
  background: ${props => props.theme.colors.blue};
  border: 1px solid ${props => props.theme.colors.grey};
  border-radius: 0.25rem;
  cursor: pointer;

  &:hover{
    opacity: 0.7;
    transition: all ease-out .45s;
  }

  ${props => props.lg && css`
    font-size: ${props => props.theme.fontSize.lg};
  `}

  ${props => props.grey && css`
    background: ${props => props.theme.colors.grey};
  `}

  ${props => props.full && css`
    width: 100%;
`}
`

export default Button