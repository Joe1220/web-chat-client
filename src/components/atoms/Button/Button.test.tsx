import React from "react";
import { shallow } from "enzyme";

import Button from "./";

describe("Button Atom Component", () => {
  let component = null;
  let number = 0;
  const onClick = () => (number = number + 1);

  it("render correctly", () => {
    component = shallow(<Button onClick={onClick}>Button</Button>);
  });
  it("matches snapshot", () => {
    expect(component).toMatchSnapshot();
  });
  it("handle Click", () => {
    component.simulate("click");
    expect(number).toBe(1);
  });
});
