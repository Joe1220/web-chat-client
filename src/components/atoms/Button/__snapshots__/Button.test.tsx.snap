// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Atom Component matches snapshot 1`] = `ShallowWrapper {}`;
